src/util/ValueSizeGetters.h

```diff
@@ -18,19 +18,20 @@ Concept for `ValueSizeGetter`. Must be default- initializable, regular invocable
 with the given value type as const l-value reference and return a `MemorySize`.
 */
 template <typename T, typename ValueType>
-concept ValueSizeGetter =
+CPP_concept ValueSizeGetter =
     std::default_initializable<T> &&
     RegularInvocableWithSimilarReturnType<T, MemorySize, const ValueType&>;
 
 // A templated default value size getter. Can be specialized for additional
 // custom types.
-template <typename T>
+template <typename T, typename = void>
 struct DefaultValueSizeGetter;
 
 // Trivially copyable types do not own heap memory (otherwise they are
 // incorred), so we just use sizeof.
 template <typename T>
-requires std::is_trivially_copyable_v<T> struct DefaultValueSizeGetter<T> {
+struct DefaultValueSizeGetter<
+    T, std::enable_if_t<std::is_trivially_copyable_v<T>>> {
```

Comment:
I'll have to check the usages here whether we can get rid of the second argument via a `using` alias.